src/modules/api/user/link/link.service.ts

```diff
@@ -26,4 +29,11 @@ export class LinkService {
     const result = await newLink.save();
     return result as ILink;
   }
+
+  async delete(linkId: string): Promise<ILink> {
+    const link = await this.linkModel.findOne({ _id: linkId });
```

Comment:
I think its better to use findById instead of findOne. In some cases where linkId is undefined, we might get arbitary results with findOne.

Reply:
Internally both are same, and linkId will never be undefined in this case. But I'll change.